src/GraphType.jl

```diff
@@ -275,7 +275,11 @@ mutable struct Graph
                 get!(req, pdict[uuid]) do; VersionSpec() end
             end
             # Translate the requirements into bit masks
-            req_msk = Dict(p1 => (pvers[p1][1:(spp[p1]-1)] .∈ vs) for (p1,vs) in req)
+            req_msk = Dict{Int,BitVector}()
+            for (p1, vs) in req
+                pv = pvers[p1]
+                req_msk[p1] = [pv[i] ∈ vs for i in 1:(spp[p1]-1)]
```

Comment:
If this is performance-critical, it may be worth trying not to go through the construction of a `Vector{Bool}` and then converting it to a `BitArray`. This could be easily done with the generator syntax `BitArray(pv[i] ∈ vs for i in 1:(spp[p1]-1)`. That however is optimized for larger arrays probably, so another option would be using a standard `BitArray` constructor and then a loop.

Reply:
I'll do some measurements, however, I think it was just the broadcast call that was the significant time thief.